dist/models/APIManager.js

```diff
@@ -0,0 +1,18 @@
+export class APIManager{
+  getAllExpensesByGroup(from = '',to = ''){
+    let d1 = (from.length > 1) ? `&d1=${from}` : ''
+    let d2 = (to.length > 1) ? `&d2=${to}` : ''
+    return $.get(`/expenses/?grouping=true${d1}${d2}`)
+  }
+  getAllGroups(){
+    return $.get('/categories')
+  }
+  addExpence(data){
+    return $.post('/new', data)
+  }
+  validator(data){
+    for (const [key, value] of Object.entries(data)) {
```

Comment:
I don't see the point of using the Object.entries method if you don't really use the key.... you could just do -

for (const key in data) { if(data[key].length<1) throw {status: 999} }